codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -352,30 +352,31 @@ let handleBlockRangeResponse = (state, ~chain, ~response: ChainWorker.blockRange
 
     let hasArbQueueEvents = state.chainManager->ChainManager.hasChainItemsOnArbQueue(~chain)
     let hasNoMoreEventsToProcess =
-      chainFetcher->ChainFetcher.hasNoMoreEventsToProcess(~hasArbQueueEvents)
+      updatedChainFetcher->ChainFetcher.hasNoMoreEventsToProcess(~hasArbQueueEvents)
 
     let latestProcessedBlock = if hasNoMoreEventsToProcess {
-      PartitionedFetchState.getLatestFullyFetchedBlock(chainFetcher.fetchState).blockNumber->Some
+      PartitionedFetchState.getLatestFullyFetchedBlock(
+        updatedChainFetcher.fetchState,
+      ).blockNumber->Some
     } else {
-      chainFetcher.latestProcessedBlock
-    }
-
-    if currentBlockHeight <= heighestQueriedBlockNumber {
-      if !ChainFetcher.isFetchingAtHead(chainFetcher) {
-        chainFetcher.logger->Logging.childInfo(
-          "All events have been fetched, they should finish processing the handlers soon.",
-        )
-      }
+      updatedChainFetcher.latestProcessedBlock
     }
 
     let partitionsCurrentlyFetching =
-      chainFetcher.partitionsCurrentlyFetching->Set.Int.remove(partitionId)
+      updatedChainFetcher.partitionsCurrentlyFetching->Set.Int.remove(partitionId)
 
     let updatedChainFetcher = {
-      ...chainFetcher,
+      ...updatedChainFetcher,
       partitionsCurrentlyFetching,
       latestProcessedBlock,
-      numBatchesFetched: chainFetcher.numBatchesFetched + 1,
+      numBatchesFetched: updatedChainFetcher.numBatchesFetched + 1,
+    }
+
+    let wasFetchingAtHead = ChainFetcher.isFetchingAtHead(chainFetcher)
+    let isCurrentlyFetchingAtHead = ChainFetcher.isFetchingAtHead(updatedChainFetcher)
+
+    if !wasFetchingAtHead && isCurrentlyFetchingAtHead {
+      updatedChainFetcher.logger->Logging.childInfo("All events have been fetched")
```

Comment:
I'm not sure if we even want to keep this log. But now it should calculate correctly at least. I also removed the second half of the log as it doesn't make full sense.